pkg/kv/kvserver/replica_raftstorage.go

```diff
@@ -566,39 +567,33 @@ func (r *Replica) applySnapshot(
 			return err
 		}
 	}
+
+	if len(inSnap.externalSSTs)+len(inSnap.sharedSSTs) == 0 && /* simple */
+		inSnap.SSTSize <= snapshotIngestAsWriteThreshold.Get(&st.SV) /* small */ {
+		ingested = false
```

Comment:
nit: the name is slightly misleading because the variable is true before we ingest. Consider `ingested -> ingest` or `applyAsIngest`.